### PR TITLE
WIP: make operator nodes support args

### LINKED ISSUE #44: FIX: allow for args
Some functions just won't do `**kwargs` so we need a syntax for providing args. This may end up a bit tricky as we need to not only map data from the events into args but also from `*args` provided in the init of the stream. We can provide the `args` positions as keys in the `input_info` dict as integers are not valid keys in a function.
Potential fixes pieces:
1. `input_info = {0: 'img', 'I0': 'I0'}` This maps the data in the `img` key to the 0th position in args yielding `func('img', 'I0')`
1. `input_info = {0: 'img', 1: 'I0'}` with `args=('b', )` we assume that we can fill any missing args positions from the `input_info` with stuff from the stream `args` thus `func('img', 'b', 'I0')`. Note that this means we must have at least as many `args` provided as there are gaps in our `input_info`. If `input_info = {0: 'img', 10: 'temperature'}` we better have at least 9 things in `args`.

@ordirules @danielballan thoughts?

### PR DESCRIPTION
**Pull request by CJ-Wright**
In a nice side note this may also help us use the star/dstar decorators less
Closes #44

### CODE REVIEW

File: shed/event_streams.py [hunk 1]
```diff
@@ -539,26 +544,29 @@ class map(EventStream):
...
                  full_event=False,
                  output_info=None, input_info=None,
                  **kwargs):
         self.func = func
-        self.kwargs = kwargs
 
         EventStream.__init__(self, child, output_info=output_info,
                              input_info=input_info, **kwargs)
+        self.kwargs = kwargs
+        self.args = args
```
Comment (ordirules) [inline shed/event_streams.py]: add `args` and `kwargs` to docstring maybe?

**Comment by CJ-Wright** [comment]
I think I broke accumulate

**Comment by CJ-Wright** [comment]
@ordirules ready for more review

File: shed/event_streams.py [hunk 2]
```diff
@@ -577,11 +601,15 @@ class filter(EventStream):
         further in the pipeline
     child: EventStream instance
         The source of the data
+    *args: tuple
+        args to be passed to the function
```
Comment (ordirules) [inline shed/event_streams.py]: maybe add:
they are appended to the args supplied by `input_info`

**Comment by CJ-Wright** [comment]
Accumulate in streams doesn't support args or kwargs so I was hoping to not support them here

**Comment by CJ-Wright** [comment]
Accumulate is no longer broken for the record

**Pull request merged**